iOS/DuckDuckGo/AutomationServer.swift

```diff
@@ -125,108 +201,103 @@ final class AutomationServer {
         let navigateUrl = URL(string: navigateUrlString)!
         self.main.loadUrl(navigateUrl)
-        self.respond(on: connection, response: "done")
+        return .success("done")
     }
 
-    func execute(on connection: NWConnection, url: URLComponents) {
+    func execute(url: URLComponents) async -> ConnectionResult {
         let script = getQueryStringParameter(url: url, param: "script") ?? ""
         var args: [String: String] = [:]
         // json decode args if present
         if let argsString = getQueryStringParameter(url: url, param: "args") {
             guard let argsData = argsString.data(using: .utf8) else {
-                self.respondError(on: connection, error: "Unable to decode args")
-                return
+                return .failure(.jsonEncodingFailed)
             }
             do {
                 let jsonDecoder = JSONDecoder()
                 args = try jsonDecoder.decode([String: String].self, from: argsData)
             } catch {
-                self.respondError(on: connection, error: error.localizedDescription)
-                return
+                Logger.automationServer.error("Failed to decode args: \(error)")
+                return .failure(.jsonEncodingFailed)
             }
         }
-        Task {
-            await self.executeScript(script, args: args, on: connection)
-        }
+        return await self.executeScript(script, args: args)
     }
 
-    func getWindowHandle(on connection: NWConnection, url: URLComponents) {
+    func getWindowHandle(url: URLComponents) -> ConnectionResult {
         let handle = self.main.currentTab
         guard let handle else {
-            self.respondError(on: connection, error: "no window")
-            return
+            return .failure(.noWindow)
         }
-        self.respond(on: connection, response: handle.tabModel.uid)
+        return .success(handle.tabModel.uid)
     }
 
-    func getWindowHandles(on connection: NWConnection, url: URLComponents) {
+    func getWindowHandles(url: URLComponents) -> ConnectionResult {
         let handles = self.main.tabManager.model.tabs.map({ tab in
             let tabView = self.main.tabManager.controller(for: tab)!
             return tabView.tabModel.uid
         })
 
         if let jsonData = try? JSONEncoder().encode(handles),
            let jsonString = String(data: jsonData, encoding: .utf8) {
-           self.respond(on: connection, response: jsonString)
+           return .success(jsonString)
         } else {
-            // Handle JSON encoding failure
-            self.respondError(on: connection, error: "Failed to encode response")
+            return .failure(.jsonEncodingFailed)
         }
     }
 
-    func closeWindow(on connection: NWConnection, url: URLComponents) {
+    func closeWindow(url: URLComponents) -> ConnectionResult {
         self.main.closeTab(self.main.currentTab!.tabModel)
-        self.respond(on: connection, response: "{\"success\":true}")
```

Comment:
I basically removed all this self.respond to make them all return ConnectionResult to ensure the control flow was followed correctly.